src/compiler_internals.md

```diff
@@ -0,0 +1,37 @@
+# Compiler Internals
```

Comment:
We should link to this in the [here](https://github.com/alantech/alan/blob/main/compiler/README.md)

Reply:
Once this PR is landed we can :)